### PR TITLE
externalize admin node commands

### PR DESCRIPTION
**Pull request by juanvallejo**
Updates `pkg/oc/cli/admin/node` commands to use external clients

cc @soltysh

### CODE REVIEW

File: pkg/oc/cli/admin/node/listpods.go [hunk 1]
```diff
@@ -32,90 +30,56 @@ func (l *ListPodsOptions) Run() error {
...
-	return kerrors.NewAggregate(errList)
-}
 
-func (l *ListPodsOptions) runListPods(node *kapi.Node, printer kprinters.ResourcePrinter) error {
-	labelSelector, err := labels.Parse(l.Options.PodSelector)
-	if err != nil {
-		return err
+		fmt.Fprintf(successMessage, "Listing matched pods on node %q:\n", node.ObjectMeta.Name)
+		for _, pod := range podList.Items {
+			fmt.Fprintf(successMessage, " - %s\n", pod.Name)
```
Comment (juanvallejo) [inline pkg/oc/cli/admin/node/listpods.go]: @soltysh this used the "humanreadable" printer before. Now it just uses the "success" message of the new printer stack. We can either leave it as it currently is, and print only pod names for each node, or try to mimic the old "humanreadable" output and print a few columns per pod (still using success output), similar to how it's being done in the policy commands PR: https://github.com/openshift/origin/pull/20810/files#diff-d16c3a1720bc945e401df0ff7c782325R268

File: pkg/oc/cli/admin/node/evacuate.go [hunk 2]
```diff
@@ -74,7 +73,7 @@ func (e *EvacuateOptions) Run() error {
 	return kerrors.NewAggregate(errList)
 }
 
-func (e *EvacuateOptions) RunEvacuate(node *kapi.Node) error {
+func (e *EvacuateOptions) RunEvacuate(node *corev1.Node) error {
```
Comment (soltysh) [inline pkg/oc/cli/admin/node/evacuate.go]: Change that alias `e` -> `o`, so be consistent with other commands.

File: pkg/oc/cli/admin/node/listpods.go (hunk 1 above)
Comment (soltysh) [reply pkg/oc/cli/admin/node/listpods.go]: Do the policy-like output.

File: pkg/oc/cli/admin/node/node_options.go [hunk 3]
```diff
@@ -8,32 +8,31 @@ import (
...
 
-	Mapper  meta.RESTMapper
-	Typer   runtime.ObjectTyper
-	Printer func(mapping *meta.RESTMapping, withNamespace bool) (kprinters.ResourcePrinter, error)
+	DefaultNamespace string
+	KubeClient       kubernetes.Interface
 
-	CmdPrinter       kprinters.ResourcePrinter
+	Mapper meta.RESTMapper
+	Typer  runtime.ObjectTyper
```
Comment (soltysh) [inline pkg/oc/cli/admin/node/node_options.go]: Do we still need `Mapper` and `Typer`?

File: pkg/oc/cli/admin/node/node_options.go [hunk 4]
```diff
@@ -84,27 +78,27 @@ func (n *NodeOptions) Complete(f kcmdutil.Factory, c *cobra.Command, args []stri
 
 	n.Builder = f.NewBuilder()
 	n.DefaultNamespace = defaultNamespace
-	n.ExternalKubeClient = externalkc
-	n.KubeClient = kc
 	n.Mapper = mapper
 	n.Typer = typer
 	n.NodeNames = []string{}
-	n.CmdPrinter = cmdPrinter
 	n.CmdPrinterOutput = false
```
Comment (soltysh) [inline pkg/oc/cli/admin/node/node_options.go]: Is this used?

File: pkg/oc/cli/admin/node/node_options.go [hunk 5]
```diff
@@ -125,14 +119,14 @@ func (n *NodeOptions) Validate(checkNodeSelector bool) error {
 	return kerrors.NewAggregate(errList)
 }
 
-func (n *NodeOptions) GetNodes() ([]*kapi.Node, error) {
+func (n *NodeOptions) GetNodes() ([]*corev1.Node, error) {
```
Comment (soltysh) [inline pkg/oc/cli/admin/node/node_options.go]: `n` -> `o`

File: pkg/oc/cli/admin/node/listpods.go (hunk 1 above)
Comment (juanvallejo) [reply pkg/oc/cli/admin/node/listpods.go]: Thanks, done

**Comment by juanvallejo** [comment]
@soltysh example output from this PR:

```bash
$ oc adm manage-node --list-pods localhost

Listing matched pods on node "localhost":

NAMESPACE                      NAME                                                  AGE
default                        pod/docker-registry-1-g648z                           1h
```

**Comment by juanvallejo** [comment]
@soltysh thanks, review comments addressed

**Comment by soltysh** [comment]
> @soltysh example output from this PR:

Why the sample output contains unnecessary empty lines? We should be consistent with other commands when it comes to empty lines. These look unnecessary.

File: pkg/oc/cli/admin/node/evacuate.go [hunk 6]
```diff
@@ -83,42 +82,46 @@ func (e *EvacuateOptions) RunEvacuate(node *kapi.Node) error {
...
 		return nil
 	}
-	rcs, err := e.Options.KubeClient.Core().ReplicationControllers(metav1.NamespaceAll).List(metav1.ListOptions{})
+	rcs, err := o.Options.KubeClient.CoreV1().ReplicationControllers(metav1.NamespaceAll).List(metav1.ListOptions{})
 	if err != nil {
 		return err
 	}
 
-	rss, err := e.Options.KubeClient.Extensions().ReplicaSets(metav1.NamespaceAll).List(metav1.ListOptions{})
+	rss, err := o.Options.KubeClient.Extensions().ReplicaSets(metav1.NamespaceAll).List(metav1.ListOptions{})
```
Comment (soltysh) [inline pkg/oc/cli/admin/node/evacuate.go]: We should switch to `Apps` for all workload controllers, this includes ReplicaSet and DaemonSet.

Comment (soltysh) [reply pkg/oc/cli/admin/node/evacuate.go]: And use versioned `AppsV1`

File: pkg/oc/cli/admin/node/evacuate.go [hunk 7]
```diff
@@ -83,42 +82,46 @@ func (e *EvacuateOptions) RunEvacuate(node *kapi.Node) error {
...
 	}
 
-	dss, err := e.Options.KubeClient.Extensions().DaemonSets(metav1.NamespaceAll).List(metav1.ListOptions{})
+	dss, err := o.Options.KubeClient.Extensions().DaemonSets(metav1.NamespaceAll).List(metav1.ListOptions{})
 	if err != nil {
 		return err
 	}
 
-	jobs, err := e.Options.KubeClient.Batch().Jobs(metav1.NamespaceAll).List(metav1.ListOptions{})
+	jobs, err := o.Options.KubeClient.Batch().Jobs(metav1.NamespaceAll).List(metav1.ListOptions{})
```
Comment (soltysh) [inline pkg/oc/cli/admin/node/evacuate.go]: `BatchV1`

Comment (soltysh) [reply pkg/oc/cli/admin/node/evacuate.go]: the others are deprecated

File: pkg/oc/cli/admin/node/listpods.go [hunk 8]
```diff
@@ -22,100 +28,179 @@ type ListPodsOptions struct {
...
 		if err != nil {
-			// Don't bail out if one node fails
 			errList = append(errList, err)
+			continue
+		}
+
+		nodeMessage := fmt.Sprintf("Listing matched pods on node %q:\n", node.ObjectMeta.Name)
+
+		if humanOutput {
+			fmt.Fprintf(o.Options.ErrOut, "\n%s\n", nodeMessage)
```
Comment (soltysh) [inline pkg/oc/cli/admin/node/listpods.go]: Drop these new lines, like I've mentioned else where.

**Comment by juanvallejo** [comment]
@soltysh 
> Why the sample output contains unnecessary empty lines? We should be consistent with other commands when it comes to empty lines. These look unnecessary.

Part of the reason was for backwards compatibility. The old output looked the same.
Also, it might be easier to read lists of pods for multiple nodes with the additional empty lines. I will update the spacing so that the output for multiple nodes looks like this:

```bash
$ oc adm manage-node --list-pods localhost
Listing matched pods on node "localhost":
NAMESPACE                      NAME                                                  AGE
default                        pod/docker-registry-1-g648z                           1h

Listing matched pods on node "node2":
NAMESPACE                      NAME                                                  AGE
default                        pod/docker-registry-1-g648z                           1h

Listing matched pods on node "node3":
NAMESPACE                      NAME                                                  AGE
default                        pod/docker-registry-1-g648z                           1h
```

**Pull request merged**